docs/setup.asciidoc

```diff
@@ -31,4 +36,42 @@ With this you enable every agent component including ASP.NET Core tracing, monit
 
 In case you only reference the `Elastic.Apm.AspNetCore` package, you won't find the `UseAllElasticApm`. Instead you need to use the `UseElasticApm()` method from the `Elastic.Apm.AspNetCore` namespace. This method turns on ASP.NET Core tracing, and gives you the opportunity to manually turn on other components. By default it will only trace ASP.NET Core requests - No HTTP request tracing, database call tracing or any other tracing component will be turned on.
 
-In case you only want to use the <<public-api>>, you don't need to do any initialization, you can simply start using the API and the agent will send the data to the APM Server.
\ No newline at end of file
+In case you only want to use the <<public-api>>, you don't need to do any initialization, you can simply start using the API and the agent will send the data to the APM Server.
+
+
+[float]
+[[setup-asp-net]]
+=== ASP.NET
+
+For ASP.NET (Full .NET Framework), once you've referenced the `Elastic.Apm.AspNetFullFramework` package,
+you can enable auto instrumentation by including the `ElasticApmModule` IIS Module in your application's `web.config`: 
+[source,xml]
+----
+<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+    <system.webServer>
+        <modules>
+            <add name="ElasticApmModule" type="Elastic.Apm.AspNetFullFramework.ElasticApmModule" />
```

Comment:
I think the order of the modules matter in case there are more, right? Maybe it'd be worth pointing out in the text below that the `ElasticAPMModule` should be the 1. one, otherwise everything that is before the `ElasticAPMModule` won't be measured as part of the transaction.

Reply:
I agree that if placing module first in `web.config` would have had the same effect as placing ASP.NET Core middleware first then documenting it would be important. But unfortunately in this case in more complicated than that, What we are actually interested in is to be **first** to be called on `BeginRequest` event but to be **last** to be called on `EndRequest` event and as described [here for example](https://stackoverflow.com/questions/14973667/can-i-be-sure-that-httpmodules-are-executed-in-the-order-in-which-they-are-liste/14974151#14974151
) it seems to be impossible with just one module.

We can have two modules: `ElasticApmModuleBeginRequest` (goes the first in web.config) and `ElasticApmModuleEndRequest` (goes the last in web.config) but I think it's a complication that is not worth the benefit.

Even though it sounds counter-intuitive it might actually be better to place our module last in `web.config` so it's called last on `EndRequest` event. The reasoning is that if there's any heavy work done by modules when request comes in this work will be around authentication and authorization and those have their own evens that are handled after `BeginRequest` so our module will account for them even if it's called last on `BeginRequest` event. On the other hand heavy work that modules might do when response is sent out (HTML links rewriting, JavaScript injection) is done on `EndRequest` event so if we want to account for that work then our module should be the last to be called on `EndRequest` event which is possible only if it's the last in web.config.

But it gets even more complicated than that... Regarding the statement above:
> (HTML links rewriting, JavaScript injection is done on `EndRequest`

Unfortunately even this is not exactly how modules might do things - for example here are two completely different ways to implement a module that adds content to HTTP response body:
1. [Using `OnEndRequest` event](https://stackoverflow.com/questions/8678424/how-to-inject-html-before-closing-html-tag-in-a-httpmodule)
2. [Using only `BeginReques` event and `HttpApplication.Context.Response.Filter`](https://stackoverflow.com/questions/33391083/injecting-javascript-on-every-page-in-asp-net)

I'm not sure if placing our module after the one that uses `HttpApplication.Context.Response.Filter` will account for work done in custom response filter.

Seeing that the whole issue is quite complicated - do we want to recommend where to place our module relative to the other modules?

Reply:
I see... yes I agree, it's best to just leave the doc as it is and maybe think about this based on feedback. 👍